src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -478,31 +481,45 @@ private void StartTelemetryIfFunctionInvocation(IDictionary<string, object> stat
                     IEnumerable<Activity> links = allScopes.GetValueOrDefault<IEnumerable<Activity>>("Links");
                     var activities = links as Activity[] ?? links?.ToArray();
 
-                    if (activities != null && activities.Length == 1)
+                    if (activities != null)
                     {
-                        operation = _telemetryClient.StartOperation<RequestTelemetry>(activities[0]);
-                        operation.Telemetry.Name = functionName;
+                        if (activities.Length == 1)
+                        {
+                            operation = _telemetryClient.StartOperation<RequestTelemetry>(activities[0]);
+                            operation.Telemetry.Name = functionName;
+                        }
+                        else
+                        {
+                            operation = CreateRequestFromLinks(activities, functionName);
+                        }
+
+                        if (this.TryGetAverageTimeInQueueForBatch(activities, operation.Telemetry.Timestamp, out long enqueuedTime))
+                        {
+                            operation.Telemetry.Metrics["timeSinceEnqueued"] = enqueuedTime;
```

Comment:
is there anything special about this key? Just noticed you're using `EnqueuedTime` for the key above and `timeSinceEnqueued` here...

Reply:
this is a magical string for Application Insights UX. It will be exposed as a metric with this name + shown on the AppMap for EventHubs. `timeSinceEnqueued` is relative time and can be a metric. while `EnqueuedTime` is absolute time.

Reply:
Got it -- I assumed it was something special but just wanted to double-check.